src/app/InteractionModelEngine.cpp

```diff
@@ -1394,5 +1394,28 @@ size_t InteractionModelEngine::GetNumDirtySubscriptions() const
     return numDirtySubscriptions;
 }
 
+void InteractionModelEngine::OnFabricRemoved(FabricIndex aFabricIndex)
+{
+    // Try to send the queued events as soon as possible. If the just emitted leave event won't
+    // be sent this time, it will likely not be delivered at all for the following reasons:
+    // - removing the fabric expires all associated ReadHandlers, so all subscriptions to
+    //   the leave event will be cancelled.
+    // - removing the fabric removes all associated access control entries, so generating
+    //   subsequent reports containing the leave event will fail the access control check.
+    GetReportingEngine().ScheduleUrgentEventDeliverySync();
```

Comment:
Be aware of https://github.com/project-chip/connectedhomeip/pull/21191/files. This must be called before ACLs are removed.

Reply:
updated, thanks